webapp/platform/types/src/utilities.ts

```diff
@@ -45,3 +45,7 @@ export type Intersection<T1, T2> =
 Omit<Omit<T1&T2, keyof(Omit<T1, keyof(T2)>)>, keyof(Omit<T2, keyof(T1)>)>;
 
 export type PartialExcept<T extends Record<string, unknown>, TKeysNotPartial extends keyof T> = Partial<T> & Pick<T, TKeysNotPartial>;
+
+export function secureGetFromRecord<T>(v: Record<string, T> | undefined, key: string) {
```

Comment:
Could you move this out of the types package? It's not a utility type, so I don't think it should go there.

While we might use this for more than message attachments, it's probably fine to live in `mattermost-redux/utils/post_utils` since that's what we're most likely going to use it on